manifests/charts/istio-control/istio-discovery/templates/deployment.yaml

```diff
@@ -151,6 +151,10 @@ spec:
             value: "{{ .Values.global.istiod.enableAnalysis }}"
           - name: CLUSTER_ID
             value: "{{ $.Values.global.multiCluster.clusterName | default `Kubernetes` }}"
+{{- if not (eq .Values.global.remotePilotAddress "") }}
+          - name: ISTIOD_CUSTOM_HOST
+            value: istiod{{- if not (eq .Values.revision "") }}-{{ .Values.revision }}{{- end }}.{{ .Release.Namespace }}.svc
```

Comment:
I am a bit confused. `remotePilotAddress` can be anything, like `istiod.example.com`. How does adding `istio-....` fix it?

Reply:
Thanks for your suggestion. I realize that it is not good to overwrite the env ISTIOD_CUSTOM_HOST. And I fix this problem through a new way.